app/controllers/reports/regions_controller.rb

```diff
@@ -34,10 +34,21 @@ def show
     @with_ltfu = with_ltfu?
 
     @children = @region.reportable_children
+
+    repo = Reports::Repository.new(@children, periods: @period, with_exclusions: report_with_exclusions?)
```

Comment:
The Repository here grabs just the data for the `@period` we want for the children data, as opposed to RegionService which will always grab many months of data. More importantly, Repository caches at the region/period level, while RegionService caches the [entire range of data](https://github.com/simpledotorg/simple-server/blob/21abf030107f6751c7285e9cd4467355a4c2341a/app/services/control_rate_service.rb#L32-L34) available for any region.

Reply:
[nit] `children_data_repository = ...`?